internal/translayer/dbuser/dbuser.go

```diff
@@ -0,0 +1,57 @@
+package dbuser
+
+import (
+	"context"
+
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/translayer"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/controller/atlas"
+)
+
+type Service struct {
+	admin.DatabaseUsersApi
+}
+
+func NewService(ctx context.Context, provider atlas.Provider, secretRef *types.NamespacedName, log *zap.SugaredLogger) (*Service, error) {
+	client, err := translayer.NewVersionedClient(ctx, provider, secretRef, log)
+	if err != nil {
+		return nil, err
+	}
+	return &Service{DatabaseUsersApi: client.DatabaseUsersApi}, nil
+}
+
+func (dus *Service) Get(ctx context.Context, db, projectID, username string) (*User, error) {
+	atlasDBUser, _, err := dus.GetDatabaseUser(ctx, projectID, db, username).Execute()
+	if err != nil {
+		if admin.IsErrorCode(err, atlas.UsernameNotFound) {
+			return nil, nil
```

Comment:
how does the consumer know that the given username was not found? `nil, nil` is ambiguous.

Reply:
We could translate the errors, at least the ones the consumer cares about.

There are 2 options usually in gets7finds for an object, **not found is fine** and a nil or empty object is returned in such case or **not found is an error** that is not usually expected. In this case I chose the first option, but as said, the explicit error might be better.